### ISSUE TITLE
conda skeleton pypi --recursive failing on new install

### ISSUE THREAD

**Issue by bldrvnlw** [conda/conda-build#1741]
Having installed conda and conda build following the tutorial instructions attempts to build with skeleton pypi --recursive fail because it can find file:   ```conda-bld/skeleton_1486539405202/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python2.7/distutils/core.py'```

The example shown here is conda skeleton pypi --recursive flake8
```
> conda skeleton pypi --recursive flake8
Using url https://pypi.python.org/packages/47/64/382631de5fd8dab367bedeff6b5b55fd9a7c883daa44f4032636e2d203ca/flake8-3.3.0.tar.gz (131 KB) for flake8.
Downloading flake8
Using cached download
Unpacking flake8...
done
working in /tmp/tmp0B3timconda_skeleton_flake8-3.3.0.tar.gz

The following NEW packages will be INSTALLED:

    openssl:    1.0.2k-0     
    pip:        9.0.1-py27_1 
    python:     2.7.13-0     
    pyyaml:     3.12-py27_0  
    readline:   6.2-2        
    setuptools: 27.2.0-py27_0
    sqlite:     3.13.0-0     
    tk:         8.5.18-0     
    wheel:      0.29.0-py27_0
    yaml:       0.1.6-0      
    zlib:       1.2.8-3      

Applying patch: '/tmp/tmp0B3timconda_skeleton_flake8-3.3.0.tar.gz/pypi-distutils.patch'
patching file core.py
Hunk #1 succeeded at 168 with fuzz 2 (offset 2 lines).
Using url https://pypi.python.org/packages/7c/69/c2ce7e91c89dc073eb1aa74c0621c3eefbffe8216b3f9af9d3885265c01c/configparser-3.5.0.tar.gz (39 KB) for configparser.
Downloading configparser
Unpacking configparser...
done
working in /tmp/tmpC43dWqconda_skeleton_configparser-3.5.0.tar.gz

Traceback (most recent call last):
  File "/home/bbmri/miniconda2/bin/conda-skeleton", line 6, in <module>
    sys.exit(conda_build.cli.main_skeleton.main())
  File "/home/bbmri/miniconda2/lib/python2.7/site-packages/conda_build/cli/main_skeleton.py", line 65, in main
    return execute(sys.argv[1:])
  File "/home/bbmri/miniconda2/lib/python2.7/site-packages/conda_build/cli/main_skeleton.py", line 61, in execute
    api.skeletonize(package, args.repo, output_dir=args.output_dir, recursive=args.recursive, version=args.version, config=config)
  File "/home/bbmri/miniconda2/lib/python2.7/site-packages/conda_build/api.py", line 209, in skeletonize
    recursive=recursive, config=config, **kwargs)
  File "/home/bbmri/miniconda2/lib/python2.7/site-packages/conda_build/skeletons/pypi.py", line 421, in skeletonize
    noprompt, packages, extra_specs, config=config, setup_options=setup_options)
  File "/home/bbmri/miniconda2/lib/python2.7/site-packages/conda_build/skeletons/pypi.py", line 688, in get_package_metadata
    config=config)
  File "/home/bbmri/miniconda2/lib/python2.7/site-packages/conda_build/skeletons/pypi.py", line 948, in get_pkginfo
    run_setuppy(src_dir, tempdir, python_version, extra_specs=extra_specs, config=config, setup_options=setup_options)
  File "/home/bbmri/miniconda2/lib/python2.7/site-packages/conda_build/skeletons/pypi.py", line 1008, in run_setuppy
    'distutils', 'core.py-copy'))
  File "/home/bbmri/miniconda2/lib/python2.7/shutil.py", line 130, in copy2
    copyfile(src, dst)
  File "/home/bbmri/miniconda2/lib/python2.7/shutil.py", line 82, in copyfile
    with open(src, 'rb') as fsrc:
IOError: [Errno 2] No such file or directory: u'/home/bbmri/miniconda2/conda-bld/skeleton_1486539405202/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python2.7/distutils/core.py'
```

**Comment by bldrvnlw** [comment]
By the way this error occurs with other pypi packages. Our original intent was to package the fastr pypi package https://pypi.python.org/pypi/fastr, when that didn't work we tried flake8, the example given above.  

Are there any steps missing in the tutorial miniconda2 installation for build (https://conda.io/docs/build_tutorials/pkgs.html) that might give rise to the errors seen?

**Comment by msarahan** [comment]
I don't think we test the --recursive flag right now (we certainly should, and I'll do that soon).  Does it work without the recursive flag?  I think it should be recursive anyway?  That flag predates me and I'm not honestly sure what purpose it serves.

**Comment by bldrvnlw** [comment]
Without --recursive it produced the bld/build and meta.yaml files.  I thought that the idea with --recursive was to include all the dependencies in the final package. 

> The --recursive flag will recursively create recipes for any dependencies of the package in the same directory. When conda build runs and a dependency in the recipe is missing, but it finds a recipe for the package in the same directory, it will recursively build the dependency automatically.

from https://www.continuum.io/blog/developer/advanced-features-conda-part-2

**Comment by msarahan** [comment]
fixed by #1754 

Note that I still needed to add some extra dependencies to make the flake8 skeleton work.  In particular, mccabe or some other flake8 dependency seemed to hard-pin setuptools to a new version, which introduced dependencies on packaging and appdirs.  We don't have packaging yet.  I added a PR internally to add it, but conda-build can't add this dependency until we have packaging in defaults.  See comment at https://github.com/conda/conda-build/pull/1754/files#diff-c61365ff02198f8397cfa6e399b6ef8bR976
